src/utils/ao.ts

```diff
@@ -218,3 +219,40 @@ export function createAoSigner(signer: ContractSigner): AoSigner {
 
   return aoSigner;
 }
+
+// using passThrough to require the minimum fields and allow others (eg TotalSupply, Logo, etc)
+export const AntStateSchema = z
+  .object({
+    Name: z.string(),
+    Ticker: z.string(),
+    Owner: z.string(),
+    Controllers: z.array(z.string()),
+    Records: z.record(
+      z.string(),
+      z
+        .object({
+          transactionId: z.string(),
+          ttlSeconds: z.number(),
+        })
+        .passthrough(),
+    ),
+    Balances: z.record(z.string(), z.number()),
+  })
+  .passthrough();
+
+/**
+ * @param state
+ * @returns {boolean}
+ * @throws {z.ZodError} if the state object does not match the expected schema
+ */
+export function isAoANTState(state: object): state is AoANTState {
+  try {
+    AntStateSchema.parse(state);
+
+    return true;
+  } catch (error) {
+    // this allows us to see the path of the error in the object as well as the expected schema on invalid fields
+    Logger.default.error(error.issues);
```

Comment:
nit: pass a logger as an function parameter (you can default it to Logger.default)